investment-app/TRINV.Api/TRINV.Application/Queries/Dashboard/GetUserInvestmentsPerformanceListQuery.cs

```diff
@@ -0,0 +1,109 @@
+    }
+
+    public async Task<OperationResult<List<InvestmentPerformanceQueryModel>>> Handle(GetInvestmentPerformanceQuery request, CancellationToken cancellationToken)
+    {
+        var operationResult = new OperationResult<List<InvestmentPerformanceQueryModel>>();
+
+        // TODO: This needs to be removed when stocks API's are added.
+        if (request.InvestmentType != InvestmentType.Cryptocurrency)
+            return operationResult.ReturnWithErrorMessage(new NotFoundException());
+
+        var currentUserInvestments = await _investmentRepository
+            .GetAllWithPredicateAsync(i => i.UserId == _userContext.UserId
+                                           && i.InvestmentType == request.InvestmentType, cancellationToken);
+
+        var groupedInvestments = currentUserInvestments
+            .GroupBy(i => i.AssetId)
+            .Select(x => new
+            {
+                Symbol = x.Key,
+                Name = x.First().Name,
+                TotalQuantity = x.Sum(i => i.Quantity),
+                TotalValueInFiat = x.Sum(i => i.PurchasePrice),
+                AverageValuePerUnitInFiat = x.Sum(i => i.PurchasePrice) / x.Sum(q => q.Quantity),
+            });
+
+        Dictionary<string, decimal> currentPrices = new Dictionary<string, decimal>();
+
+        if(request.InvestmentType == InvestmentType.Cryptocurrency)
+        {
+            var currentCryptoPrices = await _mediator.Send(new GetDigitalCurrencyListQuery(), cancellationToken);
+
+            foreach (var item in groupedInvestments)
+            {
+                var currentItemPrice = decimal.Parse(currentCryptoPrices.RelatedObject
+                    .Where(s => s.Symbol == item.Symbol)
+                    .Select(p =>p.PriceUsd)
+                    .FirstOrDefault());
+
+                currentPrices.Add(item.Symbol, currentItemPrice);
+            }
+        }
+
+        var currentUserInvestmentsPerformanceList = new List<InvestmentPerformanceQueryModel>();
+
+        foreach (var investment in groupedInvestments)
+        {
+            var currentInvestmentPrice = currentPrices[investment.Symbol];
+
+            var returnRate = ((currentInvestmentPrice - investment.AverageValuePerUnitInFiat) / investment.AverageValuePerUnitInFiat) * 100;
+
+            var result = new InvestmentPerformanceQueryModel
+            {
+                AssetId = investment.Symbol,
+                Name = investment.Name,
+                TotalInitialInvestment = Math.Round(investment.TotalValueInFiat, 2),
+                TotalCurrentInvestment = Math.Round(currentInvestmentPrice, 2),
+                Rate = Math.Round(returnRate, 2)
+            };
+
+            currentUserInvestmentsPerformanceList.Add(result);
+        }
+
+        currentUserInvestmentsPerformanceList = currentUserInvestmentsPerformanceList
+            .OrderByDescending(x => x.Rate)
+            .ToList();
+
+        operationResult.RelatedObject = currentUserInvestmentsPerformanceList;
```

Comment:
You can remove  line 88 and do it like this operationResult.RelatedObject = currentUserInvestmentsPerformanceList.OrderBy(x=>X..rate);
 
I'm just mentioning it., Don't spend time i will fix it when i am adjusting the fe